src/checks/hyper-training.js

```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  description: 'A Pokémon from SuMo can only be hyper trained if level 100.',
+  filter (pkmn) {
+    return pkmn.level !== 100;
```

Comment:
I don't like this filter and test. It's quite complicated to read. We don't want to play coding golf here.

Filter on the game, imo.

Then do something like

if (pokemon.level ===  100) { 
  return true 
}
if (pokemon.hyperTrained || .. ) { 
  return false 
)

That makes it really clear what it's doing. As I read through this, it's really complicated to figure out what the logic is actually doing, it kinda feels unnecessary to make it so concise, and also we don't want to check non-gen 7 pokemon.